src/set_theory/cardinal/basic.lean

```diff
@@ -733,54 +725,23 @@ lift_monotone.map_min
 
-protected lemma le_sup_iff {ι : Type v} {f : ι → cardinal.{max v w}} {c : cardinal} :
-  (c ≤ sup f) ↔ (∀ b, (∀ i, f i ≤ b) → c ≤ b) :=
-⟨λ h b hb, h.trans (sup_le hb), λ h, h _ $ le_sup f⟩
-
 /-- The lift of a supremum is the supremum of the lifts. -/
-lemma lift_sup {ι : Type v} (f : ι → cardinal.{max v w}) :
-  lift.{u} (sup.{v w} f) = sup.{v (max u w)} (λ i : ι, lift.{u} (f i)) :=
+lemma lift_Sup {s : set cardinal} (hs : bdd_above s) : lift.{u} (Sup s) = Sup (lift.{u} '' s) :=
 begin
-  apply le_antisymm,
-  { rw [cardinal.le_sup_iff], intros c hc, by_contra h,
-    obtain ⟨d, rfl⟩ := cardinal.lift_down (not_le.mp h).le,
-    simp only [lift_le, sup_le_iff] at h hc,
-    exact h hc },
-  { simp only [cardinal.sup_le, lift_le, le_sup, implies_true_iff] }
+  apply ((le_cSup_iff' (bdd_above_image _ hs)).2 (λ c hc, _)).antisymm (cSup_le' _),
+  { by_contra h,
+    obtain ⟨d, rfl⟩ := cardinal.lift_down (not_le.1 h).le,
+    simp_rw lift_le at h hc,
+    rw cSup_le_iff' hs at h,
+    exact h (λ a ha, lift_le.1 $ hc (mem_image_of_mem _ ha)) },
+  { rintros i ⟨j, hj, rfl⟩,
+    exact lift_le.2 (le_cSup hs hj) },
 end
 
-/-- To prove that the lift of a supremum is bounded by some cardinal `t`,
-it suffices to show that the lift of each cardinal is bounded by `t`. -/
-lemma lift_sup_le {ι : Type v} (f : ι → cardinal.{max v w})
-  (t : cardinal.{max u v w}) (w : ∀ i, lift.{u} (f i) ≤ t) :
-  lift.{u} (sup f) ≤ t :=
-by { rw lift_sup, exact sup_le w }
-
-@[simp] lemma lift_sup_le_iff {ι : Type v} (f : ι → cardinal.{max v w}) (t : cardinal.{max u v w}) :
-  lift.{u} (sup f) ≤ t ↔ ∀ i, lift.{u} (f i) ≤ t :=
-⟨λ h i, (lift_le.mpr (le_sup f i)).trans h,
- λ h, lift_sup_le f t h⟩
-
-universes v' w'
-
-/--
-To prove an inequality between the lifts to a common universe of two different supremums,
-it suffices to show that the lift of each cardinal from the smaller supremum
-if bounded by the lift of some cardinal from the larger supremum.
--/
-lemma lift_sup_le_lift_sup
-  {ι : Type v} {ι' : Type v'} (f : ι → cardinal.{max v w}) (f' : ι' → cardinal.{max v' w'})
-  (g : ι → ι') (h : ∀ i, lift.{max v' w'} (f i) ≤ lift.{max v w} (f' (g i))) :
-  lift.{max v' w'} (sup f) ≤ lift.{max v w} (sup f') :=
-lift_sup_le.{max v' w'} f _ (λ i, (h i).trans $ by { rw lift_le, apply le_sup })
-
-/-- A variant of `lift_sup_le_lift_sup` with universes specialized via `w = v` and `w' = v'`.
-This is sometimes necessary to avoid universe unification issues. -/
-lemma lift_sup_le_lift_sup'
-  {ι : Type v} {ι' : Type v'} (f : ι → cardinal.{v}) (f' : ι' → cardinal.{v'})
-  (g : ι → ι') (h : ∀ i, lift.{v'} (f i) ≤ lift.{v} (f' (g i))) :
-  lift.{v'} (sup.{v v} f) ≤ lift.{v} (sup.{v' v'} f') :=
-lift_sup_le_lift_sup f f' g h
+/-- The lift of a supremum is the supremum of the lifts. -/
+lemma lift_supr {ι : Type v} {f : ι → cardinal.{w}} (hf : bdd_above (range f)) :
+  lift.{u} (supr f) = ⨆ i, lift.{u} (f i) :=
+by { unfold supr, convert lift_Sup hf, rw range_comp }
```

Comment:
I think there some api for `supr` that aviods the need for `range_comp` here.

Reply:
I'm not aware of it, maybe @YaelDillies knows?

Reply:
I was thinking of `Sup_eq_supr` but it's not useful here. Could eliminate the `convert` with:
```suggestion
by rw [supr, supr, lift_Sup hf, ←range_comp]
```